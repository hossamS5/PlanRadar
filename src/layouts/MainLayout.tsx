import React from "react";
import { Outlet } from "react-router-dom";
import Nav from "../components/molecules/Nav/Nav";

const MainLayout: React.FC = () => {
  return (
    <div className="h-full min-h-screen flex flex-col">
      <Nav />
      <main className="w-full h-full flex-1 bg-grayPalette-100 py-9">
        {<Outlet />}
      </main>
    </div>
  );
};

export default MainLayout;
