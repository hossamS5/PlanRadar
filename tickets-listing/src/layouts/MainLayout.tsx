import React from "react";
import { Outlet } from "react-router-dom";
import Nav from "../components/molecules/Nav/Nav";

const MainLayout: React.FC = () => {
  return (
    <div className="h-full">
      <Nav />
      <main className="w-full h-full bg-grayPalette-100 min-h-screen py-9">
        {<Outlet />}
      </main>
    </div>
  );
};

export default MainLayout;
