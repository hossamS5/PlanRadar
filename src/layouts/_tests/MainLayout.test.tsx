import { render } from "@testing-library/react";
import MainLayout from "../MainLayout";
import { vi } from "vitest";

vi.mock("../../assets/icons", () => ({
  PlanradarLogo: () => <svg data-testid="PlanradarLogo" />,
  UsersIcon: () => <svg data-testid="UsersIcon" />,
}));

describe("MainLayout", () => {
  it("match snapshot", () => {
    const { asFragment } = render(<MainLayout />);
    expect(asFragment()).toMatchSnapshot();
  });
});
