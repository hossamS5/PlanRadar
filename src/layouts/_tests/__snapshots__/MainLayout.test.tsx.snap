// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout > match snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen flex flex-col"
  >
    <header
      class="h-[96px] inset-x-0 top-0 z-50  text-mainTheme sticky bg-white shadow-md"
    >
      <div
        class="appContainer w-full h-full"
      >
        <nav
          class="flex items-center gap-16 justify-between w-full h-full"
        >
          <svg
            data-testid="PlanradarLogo"
          />
          <div
            class="flex items-center gap-2"
          >
            <span
              class="text-primary-700 text-sm font-medium"
            >
              Hossam Salem
            </span>
            <svg
              data-testid="UsersIcon"
            />
          </div>
        </nav>
      </div>
    </header>
    <main
      class="w-full h-full flex-1 bg-grayPalette-100 py-9"
    />
  </div>
</DocumentFragment>
`;
